packages/core/src/controllers/subscriber.ts

```diff
@@ -357,10 +361,10 @@ export class Subscriber extends ISubscriber {
     this.relayer.core.heartbeat.on(HEARTBEAT_EVENTS.pulse, () => {
       this.checkPending();
     });
-    this.relayer.provider.on(RELAYER_PROVIDER_EVENTS.connect, async () => {
```

Comment:
Why are we moving the listener binding from `provider` to `relayer` directly?

Reply:
I moved them initially because the `relayer` fires exactly these events too so it reduced a bit of overhead and gives the relayer more control over the events but just tested and the fix of the PR doesn't need that change so I reverted to the original usage